src/connector/src/source/mod.rs

```diff
@@ -42,3 +42,25 @@ pub use mock_external_table::MockExternalTableReader;
 pub use crate::source::filesystem::{S3_CONNECTOR, S3_V2_CONNECTOR};
 pub use crate::source::nexmark::NEXMARK_CONNECTOR;
 pub use crate::source::pulsar::PULSAR_CONNECTOR;
+
+pub fn is_key_belong_to_format_encode_options(key: &str, connector: String) -> bool {
+    const PREFIXES: &[&str] = &[
+        "schema.registry",
+        "schema.location",
+        "message",
+        "key.message",
+        "without_header",
+        "delimiter",
+        // AwsAuthProps
+        "region",
+        "endpoint_url",
+        "access_key",
+        "secret_key",
+        "session_token",
+        "arn",
+        "external_id",
+        "profile",
+    ];
+    PREFIXES.iter().any(|prefix| key.starts_with(prefix))
+        || (key == "endpoint" && !connector.eq_ignore_ascii_case(KINESIS_CONNECTOR))
```

Comment:
`endpoint` is also found used (and only used) in Kinesis properties, so we move `endpoint` to `format_encode_options` if the connector is not Kinesis.
```rust
pub struct KinesisCommon {
    #[serde(rename = "stream", alias = "kinesis.stream.name")]
    pub stream_name: String,
    #[serde(rename = "aws.region", alias = "kinesis.stream.region")]
    pub stream_region: String,
    #[serde(rename = "endpoint", alias = "kinesis.endpoint")]
    pub endpoint: Option<String>,
```

Reply:
* I have no experience on aws config names used for connecting to kinesis (`endpoint` / `kinesis.endpoint`) vs names used for reading schema definition from s3 (`endpoint_url`). Would rely on @tabVersion with more experience to make sure the names are listed correctly, **and with proper tests**.
* My previous comment used the word **copy** instead of **move**. I am not sure if some names in legacy `with_properties` are actually used by both `with` and `format encode`. Moving them away from `with_properties` may break things. Again, whether this is an real issue depends on the meaning / usage of specific names, which has been obscure for a long time every time we talk about `with properties`. Just bringing the idea up and would rely on your judgement.

Reply:
Studied [serde](https://serde.rs/field-attrs.html) `rename` and `alias`, and it turns out the only name shared by both is `endpoint`, which would not be moved in the case of kinesis. **And existing kinesis connector are not using `format encode` from s3. (or we need to _copy_ `endpoint` even when it is kinesis)**

All other names for reading `format encode` from s3 has already been listed above properly.

Just for clarity, this is a concrete example of the current situation:
```
create [source | table] t with (
  connector = 'kinesis',
  stream = 't',
  aws.region = 'r0',
  endpoint = 'e0',
  aws.credentials.access_key_id = 'ak0',
  aws.credentials.secret_access_key = 'sk0'
) format plain encode avro (
  schema.location = 's3://foo/bar',
  region = 'r1',
  access_key = 'ak1',
  secret_key = 'sk1'
);
```